Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM golang:1.14.2-alpine AS builder
+
+RUN apk add --update --no-cache ca-certificates bash make gcc musl-dev git openssh wget curl bzr
+
+WORKDIR /go/src/matterbuild
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
```

Comment:
Nit pick: Is this needed? Will `make build` take care of fetching the dependencies?

Reply:
Lines 7-9 is to cache those layers and make the building faster, otherwise `make build` always tries to download dependencies even though if nothing had changed in go.mod or go.mod and would never cache those layers.

Reply:
0/5 if caching is ready needed, but I'm fine with having it

Reply:
Well it kind of is needed, specially when you start using Tilt (or similar) and try to rebuild and live reload image onto the cluster, it's gonna save some precious dozen seconds each time something changes.